core/dbt/config/project.py

```diff
@@ -397,6 +398,8 @@ def create_project(self, rendered: RenderComponents) -> 'Project':
             vars_dict = cfg.vars
 
         vars_value = VarProvider(vars_dict)
+        # There will never be any project_env_vars when it's first created
+        project_env_vars: Dict[str, Any] = {}
```

Comment:
Is this true?  With the new interactive `init` you can technically put in env_vars.

Reply:
Yes. The project_env_vars only come from rendering, which happens after the object is created. So whether or not env vars exist before the dbt code starts up doesn't matter.